tests/integrations/base/test_ToolCache.py

```diff
@@ -23,6 +24,62 @@ def simple_tools(tmp_path):
     )
 
 
+def tools_for_module(tool_module_name: str) -> set:
+    """Return names of classes that subclass Tool in a module in
+    ``briefcase.integrations``, e.g. "android_sdk"."""
+    return {
+        klass_name
+        for klass_name, _ in inspect.getmembers(
+            sys.modules[f"briefcase.integrations.{tool_module_name}"],
+            lambda klass: (
+                inspect.isclass(klass) and issubclass(klass, Tool) and klass is not Tool
+            ),
+        )
+    }
+
+
+def test_toolcache_typing():
+    """Tool typing for ToolCache is correct."""
+    # Tools that are intentionally not annotated in ToolCache.
+    tools_unannotated = {"cookiecutter"}
+    # Tool names to exclude from the dynamic annotation checks; they are manually checked.
+    tool_names_skip_dynamic_check = {"app_context", "git", "xcode", "xcode_cli"}
+    # Tool classes to exclude from dynamic annotation checks.
+    tool_klasses_skip_dynamic_checks = {"DockerAppContext", "NativeAppContext"}
+
+    # Ensure defined Tool modules/classes are annotated in ToolCache.
+    for tool_module_name in briefcase.integrations.__all__:
+        if tool_module_name not in tools_unannotated:
+            assert tool_module_name in ToolCache.__annotations__.keys()
+        for tool_name in tools_for_module(tool_module_name):
```

Comment:
It's not a huge deal, but `tools_for_module` is a moderately complex method with an answer that shouldn't change; both of the invocations are occurring inside for loops. Is there any reason we shouldn't pre-compute once and store, rather than re-computing every time?

Reply:
I avoided caching mostly for simplicity. Although, `tools_for_module()` should only be called twice for any particular module.

I can add the `functools.lru_cache` decorator to `tools_for_module()` to get caching for "free" though.

Reply:
That's a bit more complexity than is warranted here. This isn't a big deal; I won't go to the mat over 2 calls to a test function that isn't *that* complex.